### PR TITLE
Flip test strategy

### PR DESCRIPTION
**Pull request by jonanswer**
add in the details on the test strategy into the documentation repo

### CODE REVIEW

File: documentation/release_procedure.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  - [SIT Process](#sit-process)
+  - [UAT Process](#uat-process)
+  - [Production Smoke Test](#production-smoke-test)
+- [Feature Flags](#feature-flags)
+- [Environments](#environments)
+- [RACI](#raci)
+
+# Executive Summary
+
+Answer Digital will manage releases to all environments including Production. The Production release cadence will directly align with the product roadmaps giving clear visibility when a new version will be available to the Trusts. Early sight of production releases will enable Trusts to plan accordingly and for system documentation to be updated.
```
Comment (heyhaleema) [inline documentation/release_procedure.md:32]:
```suggestion
Answer Digital (AD) will manage releases to all environments including Production. The Production release cadence will directly align with the product roadmaps giving clear visibility when a new version will be available to the Trusts. Early sight of production releases will enable Trusts to plan accordingly and for system documentation to be updated. 
```

File: documentation/test_strategy.md [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+  - [FLIP](#flip)
+- [Scope](#scope)
+  - [In Scope](#in-scope)
+  - [Out of Scope](#out-of-scope)
+- [Roles and Responsibilities](#roles-and-responsibilities)
+
+#
+# Executive Summary
+
+Answer Digital promotes a collaborative testing culture. This will be achieved by implementing an Acceptance Test Driven Development (ATDD) automation framework. Tests will be created in example mapping sessions and will focus on testing the behaviour of a feature through test scenarios written in plain English. This will enable non technical stakeholders to review and help create automation test scenarios. Increasing the visibility of the test scenarios will establish test transparency and shorten UAT cycles as stakeholders have reviewed and defined the test coverage. Test Scenarios will be attached to User Stories and reports will be produced to demonstrate that the agreed coverage is executed for every release.
```
Comment (heyhaleema) [inline documentation/test_strategy.md:55]:
```suggestopm
Answer Digital (AD) promotes a collaborative testing culture. This will be achieved by implementing an Acceptance Test Driven Development (ATDD) automation framework. Tests will be created in example mapping sessions and will focus on testing the behaviour of a feature through test scenarios written in plain English. This will enable non technical stakeholders to review and help create automation test scenarios. Increasing the visibility of the test scenarios will establish test transparency and shorten UAT cycles as stakeholders have reviewed and defined the test coverage. Test Scenarios will be attached to User Stories and reports will be produced to demonstrate that the agreed coverage is executed for every release.
```

File: documentation/test_strategy.md [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+
+This document will continue to be updated throughout the lifetime of the project, drawing on improved levels of knowledge and insight on requirements for ensuring quality of the delivered solutions. This information will be maintained in a shared knowledge repository, utilising Atlassian Confluence.
+
+#
+# Project Overview
+
+The London Medical Imaging & AI Centre for Value Based Healthcare have commissioned an AI programme involving a consortium of 10 NHS Trusts, with the objective of increasing efficiency and improving patients’ outcomes through advanced diagnostics and identification of personalised treatments.
+
+King’s College London and Guy’s and St Thomas’ NHS Foundation Trust are leading the consortium to develop an artificial intelligence based Federated Learning and Interoperability Platform (FLIP) for diagnostic imaging, and an Artificial Intelligence Deployment Engine (AIDE) for real world AI clinical deployment. This platform will provide the toolset for the AI Centre, enabling researchers and clinicians to transform and deploy advanced diagnostic pathways.
```
Comment (heyhaleema) [inline documentation/test_strategy.md:73]:
```suggestion
King’s College London (KCL) and Guy’s and St Thomas’ NHS Foundation Trust (GSTT) are leading the consortium to develop an artificial intelligence (AI) based Federated Learning and Interoperability Platform (FLIP), and AIDE, an AI deployment engine, for real world AI clinical deployment. These platforms will provide the toolset for the AI Centre, enabling researchers and clinicians to transform and deploy advanced diagnostic pathways.
```

File: documentation/test_strategy.md [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+
+The programme is split into the following components:
+
+- **AI Deployment Engine (AIDE)**: enabling NHS Trusts to deploy AI products in clinical practice.
```
Comment (heyhaleema) [inline documentation/test_strategy.md:77]:
```suggestion
- **AIDE**: enabling NHS Trusts to deploy AI products in clinical practice.
- **FLIP**: enabling NHS Trusts to make their data accessible, without compromising patient confidentiality to develop AI tools.
```

File: documentation/test_strategy.md [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+- **Federated Learning and Interoperability Platform (FLIP)**: enabling NHS Trusts to make their data accessible, without compromising patient confidentiality to develop AI tools.
```
Comment (heyhaleema) [inline documentation/test_strategy.md:78]:
```suggestion
- **AIDE**: enabling NHS Trusts to deploy AI products in clinical practice.
- **FLIP**: enabling NHS Trusts to make their data accessible, without compromising patient confidentiality to develop AI tools.
```

File: documentation/test_strategy.md [hunk 6]
```diff
@@ -0,0 +1,558 @@
...
+
+## Database Testing
+
+Database testing will involve checking the data integrity of any databases for both FLIP and AIDE solutions. Automated tests will be created to check that the databases adhere to the correct schema and the expected data is mapped  correctly. The tests will cover the following:
+
+- Schema / Mapping validation
+- Data transformation validation
+- Stored procedure and view testing
+- Manual verification of any PII masking
+- Non-Functional inc. Security and Benchmark testing
```
Comment (heyhaleema) [inline documentation/test_strategy.md:145]:
```suggestion
- Non-Functional, incuding Security and Benchmark testing
```

**Review by heyhaleema — APPROVED** [review]
@jonanswer Looks good! I'm going to request a quick review from @Lucrezia-Cester, too, but otherwise happy to merge.

**Comment by heyhaleema** [comment]
@Lucrezia-Cester Would you mind having a quick review of this PR? No need to spend too much time on it, but I thought it'd be good to get a second pair of eyes on the changes, too 🧐

**Comment by Lucrezia-Cester** [comment]
hI @heyhaleema, I cannot comment on whether these procedures are ok or not as I am not very familiar, however, for what it is worth, the document does sound reasonable to me
